v2/oauth/customizations/working-with-scopes.mdx

```diff
@@ -0,0 +1,77 @@
+---
+title: Working with Scopes
+hide_title: true
+---
+
+import BackendSDKTabs from "/src/components/tabs/BackendSDKTabs"
+import TabItem from '@theme/TabItem';
+
+# Working with Scopes
+
+The allowed scopes are set during the creation process of an **OAuth2 Client**.
+
+By default, our **OAuth2** implementation adds the following built-in scopes:
+- `email`: adds the `email` and `email_verified` claims into the **ID Token** and the **User Info**.
+- `phoneNumber`: adds the `phoneNumber` and `phoneNumber_verified` claims into the **ID Token** and the **User Info**
+- `emails`:  adds the `emails` claim into the **ID Token** and the **User Info**. It includes all the emails associated with that user.
+- `phoneNumbers`: adds the `phoneNumbers` claim into the **ID Token** and the **User Info**. It includes all the phone numbers associated with that user.
+- `roles`: adds the roles claim into the **ID Token** and Access Token.
+   - This will contain the roles returned by `getRolesForUser`
+   - This only works if the `UserRoles` recipe is initialized
+- `permissions`: adds the `permissions` claim into the **ID Token** and Access Token.
+   - This will contain the list of permissions obtained by concatenating the result of `getPermissionsForRole` for all roles returned by `getRolesForUser`
+   - This only works if the `UserRoles` recipe is initialized
+ 
+## Requesting Specific Scopes
+
+The client can request specific scopes by adding `scope` query param to the **Authorization URL**.
+The requested scopes have to be a subset of what is allowed for the client, otherwise the authentication request will fail.
+By default all scopes are granted to the client.
+
+## Overriding granted scopes
+
+If you want to manually modify the list of scopes that are granted to the client during the authentication flow you can do this by using overrides.
+
+
+<BackendSDKTabs>
+
+<TabItem value="nodejs">
+
+```typescript
+OAuth2Provider.init({
+    override: {
+        functions: (originalFunctions) => ({
+            ...originalFunctions,
+            getRequestedScopes: async (input) => {
+                const originallyRequestedScopes = await originalFunctions.getRequestedScopes(input);
+                const filteredScopes = originallyRequestedScopes.filter((scope) => scope !== "profile");
+                return [...filteredScopes, "custom-scope"];
+            },
+        }),
+    },
+});
+```
+
+</TabItem>
+
+<TabItem value="go">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Go SDK.
+
+:::
+
+</TabItem>
+
+<TabItem value="python">
+
+:::caution
+
+At the moment we do not have support creating OAuth2 providers in the Python SDK.
+
+:::
+
+</TabItem>
+
+</BackendSDKTabs>
```

Comment:
Need a section about how users can add custom scopes that makes sense for their application

Reply:
Can you elaborate on this one?